### PR TITLE
Add support for running sidecars container

### PR DESCRIPTION
**Pull request by radek-starburst**

### CODE REVIEW

File: charts/trino/values.yaml [hunk 1]
```diff
@@ -125,10 +125,26 @@ initContainers: {}
   #     image: busybox:1.28
   #     command: ['sh', '-c', 'echo The worker is running! && sleep 3600']
 
+sidecarContainers: {}
```
Comment (nineinchnick) [inline charts/trino/values.yaml:128]: We already document that `initContainers` can be used for sidecar containers: https://docs.starburst.io/latest/k8s/sep-configuration.html#sidecar-and-init-containers

File: charts/trino/templates/deployment-coordinator.yaml [hunk 2]
```diff
@@ -40,6 +40,9 @@ spec:
         runAsUser: {{ .runAsUser }}
         runAsGroup: {{ .runAsGroup }}
       {{- end }}
+      {{- if .Values.shareProcessNamespace.coordinator }}
```
Comment (losipiuk) [inline charts/trino/templates/deployment-coordinator.yaml:43]: why conditionally?

File: charts/trino/templates/deployment-worker.yaml [hunk 3]
```diff
@@ -44,6 +44,9 @@ spec:
         runAsUser: {{ .runAsUser }}
         runAsGroup: {{ .runAsGroup }}
       {{- end }}
+      {{- if .Values.shareProcessNamespace.worker }}
```
Comment (losipiuk) [inline charts/trino/templates/deployment-worker.yaml:47]: why conditionally?

File: charts/trino/Chart.yaml [hunk 4]
```diff
@@ -15,7 +15,7 @@ type: application
 # This is the chart version. This version number should be incremented each time you make changes
 # to the chart and its templates, including the app version.
 # Versions are expected to follow Semantic Versioning (https://semver.org/)
-version: 0.13.0
+version: 0.13.0-radek
```
Comment (losipiuk) [inline charts/trino/Chart.yaml]: revert

Comment (losipiuk) [reply charts/trino/Chart.yaml]: (I mean bump to 0.14.0)

File: charts/trino/values.yaml (hunk 1 above)
Comment (radek-starburst) [reply charts/trino/values.yaml:128]: Does it apply to trino?

File: charts/trino/templates/deployment-coordinator.yaml (hunk 2 above)
Comment (radek-starburst) [reply charts/trino/templates/deployment-coordinator.yaml:43]: To not force user to define `shareProcessNamespace`

File: charts/trino/Chart.yaml (hunk 4 above)
Comment (radek-starburst) [reply charts/trino/Chart.yaml]: To not force user to define `shareProcessNamespace`

File: charts/trino/templates/deployment-worker.yaml (hunk 3 above)
Comment (radek-starburst) [reply charts/trino/templates/deployment-worker.yaml:47]: To not force user to define `shareProcessNamespace`

Comment (losipiuk) [reply charts/trino/templates/deployment-worker.yaml:47]: isn't default always a fallback anyway?

Comment (losipiuk) [reply charts/trino/templates/deployment-worker.yaml:47]: and you would just put `shareProcessNamespace: false` if it is not redefined explicitly?

Comment (radek-starburst) [reply charts/trino/templates/deployment-worker.yaml:47]: changed.

File: README.md [hunk 5]
```diff
@@ -22,7 +22,7 @@ You can then run `helm search repo trino` to see the charts.
 Then you can install chart using:
 
 ```console
-helm install my-trino trino/trino --version 0.13.0
+helm install my-trino trino/trino --version 0.14.0
```
Comment (mosabua) [inline README.md:25]: Lets do the release update in a separate PR from the feature.

File: charts/trino/values.yaml [hunk 6]
```diff
@@ -125,10 +125,26 @@ initContainers: {}
...
+#   coordinator:
+#     - name: side-coordinator
+#       image: busybox:1.28
```
Comment (mosabua) [inline charts/trino/values.yaml:131]: is busybox really a good example?

File: charts/trino/values.yaml [hunk 7]
```diff
@@ -125,10 +125,26 @@ initContainers: {}
...
+#       imagePullPolicy: IfNotPresent
+#       command: ['sleep', '1']
+
 securityContext:
   runAsUser: 1000
   runAsGroup: 1000
 
+shareProcessNamespace:
+#  coordinator: false
```
[the remark this answers is not in the document]
Comment (radek-starburst) [reply charts/trino/values.yaml:145]: No, it is optional.

File: charts/trino/values.yaml (hunk 6 above)
Comment (radek-starburst) [reply charts/trino/values.yaml:131]: It is just an example - it depends on specific use case what sidecar's image will be used.

**Review by losipiuk — APPROVED** [review]

**Comment by losipiuk** [comment]
cc: @nineinchnick
